src/python/pants/util/collections.py

```diff
@@ -19,3 +19,10 @@ def recursively_update(d, d2):
         recursively_update(d[k], v)
         continue
     d[k] = v
+
+
+def assert_single_element(iterable):
+  """Convert `iterable` to a list, assert that the list has one element, then return the element."""
+  result = list(iterable)
+  assert(len(result) == 1)
```

Comment:
Using an `assert` for something like this isn't that friendly. Should probably actually have an error message.

Reply:
Fixed in 3be53f3!